dttools/src/mq.h

```diff
@@ -303,6 +315,22 @@ void *mq_poll_error(struct mq_poll *p);
  */
 int mq_send_buffer(struct mq *mq, buffer_t *buf);
 
+/** Stream a file descriptor across the wire.
+ *
+ * This operates in the same way as @ref mq_send_buffer, but takes
+ * a file descriptor. Note that the queue takes ownership of fd and will
+ * close it when the message is sent, so callers MUST NOT use/close fd
+ * after passing it in. This function will read from fd until reaching the
+ * end of the file. This function will not seek fd, so it's possible to send
+ * slices from within files.
+ * @param mq The message queue.
+ * @param fd The file descriptor to read.
+ * @returns 0 on success. Note that this only indicates that the message was
+ *  successfully queued. It gives no indication about delivery.
+ * @returns -1 on failure.
+ */
+int mq_send_fd(struct mq *mq, int fd);
```

Comment:
I think you want to pass in the size of the buffer as an argument here -- it's going to be a common case that the size of the file is known in advance (or it's coming from a messages of known length from the client)